### PR TITLE
feat(utilities): add text and background utilities

### LINKED ISSUE #3642: Review text color and background color utility classes
Currently we don't have much of a strategy around the colors and background colors offered in the new text and background color utility classes. I'd like to take some time and consider what variations we should allow for colors and background colors, and track that in this issue.

Some thoughts from a short conversation with design:

* Text colors should be limited to the status colors we apply to text in components with status (eg, success alert title text)
* Icons can be any text and status color. Does that mean we have icon-color and text-color classes?
* Background colors are limited to just global backgrounds. What about status backgrounds - inline alert backgrounds for example.

### PR DESCRIPTION
**Pull request by srambach**
This adds in additional text color and background color utilities.

Fixes #3642

### CODE REVIEW

File: src/patternfly/utilities/BackgroundColor/examples/BackgroundColor.md [hunk 1]
```diff
@@ -6,87 +6,155 @@ beta: true
...
-| `.pf-u-background-color-dark-100` | `*`        | Applies background-color-dark 100. |
-| `.pf-u-background-color-dark-200` | `*`        | Applies background-color-dark 200. |
-| `.pf-u-background-color-dark-300` | `*`        | Applies background-color-dark 300. |
-| `.pf-u-background-color-dark-400` | `*`        | Applies background-color-dark 200. |
+| `.pf-u-background-color-100`      | `*`        | Applies background color 100.      |
+| `.pf-u-background-color-200`      | `*`        | Applies background color 200.      |
+| `.pf-u-background-color-dark-100` | `*`        | Applies background color dark 100. |
+| `.pf-u-background-color-dark-200` | `*`        | Applies background color dark 200. |
+| `.pf-u-background-color-dark-300` | `*`        | Applies background color dark 300. |
+| `.pf-u-background-color-dark-400` | `*`        | Applies background color dark 200. |
```
Comment (mcoker) [inline src/patternfly/utilities/BackgroundColor/examples/BackgroundColor.md:153]:
```suggestion
| `.pf-u-background-color-dark-400` | `*`        | Applies background color dark 400. |
```

File: src/patternfly/utilities/BackgroundColor/examples/BackgroundColor.md [hunk 2]
```diff
@@ -6,87 +6,155 @@ beta: true
...
 | Class                             | Applied to | Outcome                            |
 | --------------------------------- | ---------- | ---------------------------------- |
-| `.pf-u-background-color-100`      | `*`        | Applies background-color 100.      |
-| `.pf-u-background-color-200`      | `*`        | Applies background-color 200.      |
-| `.pf-u-background-color-dark-100` | `*`        | Applies background-color-dark 100. |
-| `.pf-u-background-color-dark-200` | `*`        | Applies background-color-dark 200. |
-| `.pf-u-background-color-dark-300` | `*`        | Applies background-color-dark 300. |
-| `.pf-u-background-color-dark-400` | `*`        | Applies background-color-dark 200. |
+| `.pf-u-background-color-100`      | `*`        | Applies background color 100.      |
+| `.pf-u-background-color-200`      | `*`        | Applies background color 200.      |
```
Comment (mcoker) [inline src/patternfly/utilities/BackgroundColor/examples/BackgroundColor.md:149]: Looks like we're missing docs for these


```
  background-color-active-color-100: (
    background-color var(--pf-global--active-color--100)
  ),
  background-color-active-color-300: (
    background-color var(--pf-global--active-color--300)
  ),
  background-color-primary-color-200: (
    background-color var(--pf-global--primary-color--100)
  ),
```

File: src/patternfly/utilities/Text/examples/Text.md [hunk 3]
```diff
@@ -123,19 +229,18 @@ Text utility
 | `.pf-u-color-200`           | `*`        | Sets font-color to Color 200           |
 | `.pf-u-color-300`           | `*`        | Sets font-color to Color 300           |
 | `.pf-u-color-400`           | `*`        | Sets font-color to Color 400           |
-| `.pf-u-icon-color-light`    | `*`        | Sets font-color to Icon Color Light    |
-| `.pf-u-icon-color-dark`     | `*`        | Sets font-color to Icon Color Dark     |
-| `.pf-u-link-color`          | `*`        | Sets font-color to Link Color          |
 | `.pf-u-active-color-100`    | `*`        | Sets font-color to Active Color 100    |
-| `.pf-u-active-color-200`    | `*`        | Sets font-color to Active Color 200    |
-| `.pf-u-active-color-300`    | `*`        | Sets font-color to Active Color 300    |
```
Comment (mcoker) [inline src/patternfly/utilities/Text/examples/Text.md:131]: Are we removing `active-color-200/300`? If so, looks like they are still in the CSS:

https://github.com/patternfly/patternfly/blob/e3962b3df7ada7a975af955dca7ac07146169eff/src/patternfly/utilities/Text/text.scss#L98-L103

File: src/patternfly/utilities/Text/text.scss [hunk 4]
```diff
@@ -59,15 +59,39 @@ $pf-u-font-color-options: (
...
+    color var(--pf-global--link--Color--light)
+  ),
+  link-color-light-hover: (
+    color var(--pf-global--link--Color--light--hover)
+  ),
+  link-color-dark: (
+    color var(--pf-global--link--Color--dark)
+  ),
+  link-color-dark-hover: (
+    color var(--pf-global--link--Color--dark-hover)
```
Comment (mcoker) [inline src/patternfly/utilities/Text/text.scss:84]:
```suggestion
    color var(--pf-global--link--Color--dark--hover)
```

File: src/patternfly/utilities/Text/examples/Text.md [hunk 5]
```diff
@@ -123,19 +229,18 @@ Text utility
...
 | `.pf-u-active-color-400`    | `*`        | Sets font-color to Active Color 400    |
-| `.pf-u-disabled-color-100`  | `*`        | Sets font-color to Disabled Color 100  |
-| `.pf-u-disabled-color-200`  | `*`        | Sets font-color to Disabled Color 200  |
-| `.pf-u-disabled-color-300`  | `*`        | Sets font-color to Disabled Color 200  |
 | `.pf-u-primary-color-100`   | `*`        | Sets font-color to Primary Color 100   |
-| `.pf-u-primary-color-200`   | `*`        | Sets font-color to Primary Color 200   |
-| `.pf-u-secondary-color-100` | `*`        | Sets font-color to Secondary Color 100 |
```
Comment (mcoker) [inline src/patternfly/utilities/Text/examples/Text.md:138]: primary-200 and secondary-100 are still in the css

https://github.com/patternfly/patternfly/blob/e3962b3df7ada7a975af955dca7ac07146169eff/src/patternfly/utilities/Text/text.scss#L116-L121

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! Just a few small things I noticed.

File: src/patternfly/utilities/Text/examples/Text.md [hunk 6]
```diff
@@ -123,19 +229,18 @@ Text utility
 | `.pf-u-color-200`           | `*`        | Sets font-color to Color 200           |
```
Comment (mcoker) [inline src/patternfly/utilities/Text/examples/Text.md:229]: Also can you update these docs to be sentence case?

**Review by mcoker — APPROVED** [review]
fantastic! 🎉

**Review by mceledonia — APPROVED** [review]
Looks good to me!

**Review by mcarrano — COMMENTED** [review]
I will defer to @mceledonia and @mcoker on this one.

**Pull request merged**